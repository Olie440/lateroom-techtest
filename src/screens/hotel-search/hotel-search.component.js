import React from 'react';
import { connect } from 'react-redux';
import logo from '../../assets/images/logo.svg';
import './hotel-search.css';

export function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          Edit <code>src/App.js</code> and save to reload.
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React
        </a>
      </header>
    </div>
  );
}

export function mapStateToProps(state) {
  return state;
}

export default connect(mapStateToProps)(App);
