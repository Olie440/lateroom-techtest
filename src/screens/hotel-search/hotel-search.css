.hotel-search-screen {
    background: #eaeaea;
    position: absolute;
    top: 0;
    bottom: 0;
    left: 0;
    right: 0;
}

.hotel-search-screen__container {
    margin: 0 auto;
    display: grid;
    justify-content: center;
    padding-top: 10%;
    grid-template-columns: 30rem;
}

.hotel-search-screen__header {
    background: #fbe17e;
    border: 1px solid #e1ca71;
    padding: 0.5rem;
}
