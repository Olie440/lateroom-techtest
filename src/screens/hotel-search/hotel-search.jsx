import React from 'react';
import styled from 'styled-components';
import HotelList from './components/hotel-list.component';
import HotelFilter from './components/hotel-filter-menu';

export default function HotelSearchScreen() {
  return (
    <Background>
      <Container>
        <Header>
          <HotelFilter />
        </Header>
        <HotelListWrapper>
          <HotelList />
        </HotelListWrapper>
      </Container>
    </Background>
  );
}

export const Background = styled.div`
  background: #eaeaea;
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
`;

export const Container = styled.div`
  margin: 0 auto;
  display: grid;
  justify-content: center;
  padding-top: 10%;
  grid-template-columns: 30rem;
`;

export const Header = styled.div`
  background: #fbe17e;
  border: 1px solid #e1ca71;
  padding: 0.5rem;
`;

const HotelListWrapper = styled.div`
  background: #ffffff;
  border: 1px solid #cccccc;
  border-top: none;
`;
