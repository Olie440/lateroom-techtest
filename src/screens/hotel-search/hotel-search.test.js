import React from 'react';
import { shallow } from 'enzyme';
import { App } from './hotel-search.component';

it('renders without crashing', () => {
  const component = shallow(<App />);
  expect(component).toMatchSnapshot();
});
