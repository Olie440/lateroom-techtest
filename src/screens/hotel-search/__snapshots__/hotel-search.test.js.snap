// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <Connect(HotelFilter) />
    </styled.div>
    <Connect(HotelList) />
  </styled.div>
</styled.div>
`;
